cmd/nvidia-dra-controller/imex.go

```diff
@@ -312,3 +312,25 @@ func (m *ImexManager) cleanupResourceSlices() error {
 
 	return nil
 }
+
+// calculateImexChannelLimit calculates the number of IMEX channels that can be allocated to a given IMEX domain.
+// if the limit is reached, it will return an error.
+func calculateImexChannelLimit(addedDomainsCh string, imexChannelOffsets *map[string]map[string]int) error {
```

Comment:
The semantics of this should not be to "calculate an ImexChannel Limit", but rather to "set the offset where an imex domain's channels should start counting from".

Which it seems is mostly what it is doing but I haven't verified the logic yet.

Reply:
Thinking about this a bit more -- let's turn `var imexChannelOffsets map[string]map[string]int` into a type so we can hang a function off of it called `GetOffset(imexDomain) (int error)` which essentally implements the logic here but returns the offset we care about for the passed in imex domain.

Reply:
K, I was already thinking on having the `int,error` return, but the type is a good touch as well.

Reply:
done, now is a type